jscomp/runtime/caml_lexer.ml

```diff
@@ -106,25 +106,26 @@ function caml_lex_array(s) {
  *)
 let caml_lex_engine_aux : lex_tables -> int -> lexbuf -> exn -> int = fun%raw tbl start_state lexbuf exn -> {|
     // Lexing.lexbuf
-    var lex_buffer = 1;
-    var lex_buffer_len = 2;
-    var lex_start_pos = 4;
-    var lex_curr_pos = 5;
-    var lex_last_pos = 6;
-    var lex_last_action = 7;
-    var lex_eof_reached = 8;
+    var lex_buffer = 'lex_buffer';
+    var lex_buffer_len = 'lex_buffer_len';
+    var lex_start_pos = 'lex_start_pos';
+    var lex_curr_pos = 'lex_curr_pos';
+    var lex_last_pos = 'lex_last_pos';
+    var lex_last_action = 'lex_last_action';
+    var lex_eof_reached = 'lex_eof_reached';
     // Lexing.lex_tables
-    var lex_base = 0;
-    var lex_backtrk = 1;
-    var lex_default = 2;
-    var lex_trans = 3;
-    var lex_check = 4;
-    if (!tbl.lex_default) {
+    var lex_base = 'lex_base';
+    var lex_backtrk = 'lex_backtrk';
+    var lex_default = 'lex_default';
+    var lex_trans = 'lex_trans';
+    var lex_check = 'lex_check';
+    if (!tbl.processed) {
         tbl.lex_base = caml_lex_array(tbl[lex_base]);
         tbl.lex_backtrk = caml_lex_array(tbl[lex_backtrk]);
         tbl.lex_check = caml_lex_array(tbl[lex_check]);
         tbl.lex_trans = caml_lex_array(tbl[lex_trans]);
         tbl.lex_default = caml_lex_array(tbl[lex_default]);
+        tbl.processed = true;
```

Comment:
what's the rational of adding `tbl.processed`, is it a bug fix?

Reply:
This is subtle:
This code already existed before my change:
```js
tbl.lex_default = caml_lex_array(tbl[lex_default]);
```
So the code was effectively using object fields in addition to arrays.
And then the check

```js
if (!tbl.lex_default) {
```
semantically means: "I have already processed this code".
In my changes, I've made this subtle distinction explicit. The behavior has not changed.